lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -418,74 +418,73 @@ def add_exemption(post, options = {})
       # the existing logic by default. To be able to utilize this field, you must reach out to Stripe.
 
       def add_stored_credentials(post, options = {})
-        return unless options[:stored_credential] && !options[:stored_credential].values.all?(&:nil?)
+        stored_credential = options[:stored_credential]
+        return unless stored_credential && !stored_credential.values.all?(&:nil?)
 
         post[:payment_method_options] ||= {}
         post[:payment_method_options][:card] ||= {}
-        add_stored_credential_transaction_type(post, options) if options[:stored_credential_transaction_type]
-
-        stored_credential = options[:stored_credential]
         post[:payment_method_options][:card][:mit_exemption] = {}
+        card_options = post[:payment_method_options][:card]
 
         # Stripe PI accepts network_transaction_id and ds_transaction_id via mit field under card.
         # The network_transaction_id can be sent in nested under stored credentials OR as its own field (add_ntid handles when it is sent in on its own)
         # If it is sent is as its own field AND under stored credentials, the value sent under its own field is what will send.
-        post[:payment_method_options][:card][:mit_exemption][:ds_transaction_id] = stored_credential[:ds_transaction_id] if stored_credential[:ds_transaction_id]
-        post[:payment_method_options][:card][:mit_exemption][:network_transaction_id] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
-      end
+        card_options[:mit_exemption][:ds_transaction_id] = stored_credential[:ds_transaction_id] if stored_credential[:ds_transaction_id]
 
-      def add_stored_credential_transaction_type(post, options = {})
-        stored_credential = options[:stored_credential]
-        # Do not add anything unless these are present.
-        return unless stored_credential[:reason_type] && stored_credential[:initiator]
+        return unless options[:stored_credential_transaction_type] && %i(reason_type initiator).all? { |k| stored_credential.key?(k) }
```

Comment:
We should only engage the `initial_transaction_stored_credential` and `subsequent_transaction_stored_credential` methods in flows where `stored_credential_transaction_type` is passed down as `true`. This commit seems to engage them in all cases. 

Also we should be able to add a NTID when `stored_credential_transaction_type` is `false`.

Reply:
Okay, thanks for clarifying that wasn't very clear when looking at the ticket. Will change it

Reply:
Ah my bad! Basically the context for the change is that we are trying to add a new field for Stripe stored creds but we cannot turn it on for everyone since it requires a configuration on the merchant side. Therefore we should only add this new "stored_credential_transaction_type" field if the merchant passes `stored_credential_transaction_type` as `true`. 

In testing it I found that Stripe fails transactions with an NTID in 2 scenarios:

1. When `stored_credential_transaction_type` is engaged AND the mapping logic leads to us setting `stored_credential_transaction_type == 'setup_on_session'`. 
2. When `stored_credential_transaction_type` is not engaged AND `setup_future_usage= 'off_session'`. 

This 2nd scenario is currently preventing us from being able to pass the standard stored credential fields (initial_transaction, initiator, etc) for CIT. I see you have the guard for the first case but we need the guard for the second case as well